utils/bazel/llvm-project-overlay/llvm/config.bzl

```diff
@@ -94,7 +94,10 @@ llvm_config_defines = os_defines + select({
     "@bazel_tools//src/conditions:linux_aarch64": native_arch_defines("AArch64", "aarch64-unknown-linux-gnu"),
     "@bazel_tools//src/conditions:linux_ppc64le": native_arch_defines("PowerPC", "powerpc64le-unknown-linux-gnu"),
     "@bazel_tools//src/conditions:linux_s390x": native_arch_defines("SystemZ", "systemz-unknown-linux_gnu"),
-    "//conditions:default": native_arch_defines("X86", "x86_64-unknown-linux-gnu"),
```

Comment:
We can define HAVE_BUILTIN_THREAD_POINTER for all targets except `powerpc64le-linux-gnu-gcc`.

`clang --target=powerpc64le` supports `__builtin_thread_pointer`

Reply:
I've refactored and set it to apply to all platforms except for `linux_ppc64le`. Does everything still look good to you?